tsl/src/compression/api.c

```diff
@@ -236,11 +237,27 @@ compress_chunk_impl(Oid hypertable_relid, Oid chunk_relid)
 	/* Perform an analyze on the chunk to get up-to-date stats before compressing */
 	preserve_uncompressed_chunk_stats(chunk_relid);
 
-	/* aquire locks on catalog tables to keep till end of txn */
+	/* acquire locks on catalog tables to keep till end of txn */
 	LockRelationOid(catalog_get_table_id(ts_catalog_get(), HYPERTABLE_COMPRESSION),
 					AccessShareLock);
 	LockRelationOid(catalog_get_table_id(ts_catalog_get(), CHUNK), RowExclusiveLock);
 
+	DEBUG_WAITPOINT("compress_chunk_impl_start");
+
+	/*
+	 * Re-read the state of the chunk after all locks have been acquired and ensure
+	 * it is still uncompressed. Another process running in parallel might have
+	 * already performed the compression while we were waiting for the locks to be
+	 * acquired.
+	 */
+	Chunk *chunk_state_after_lock = ts_chunk_get_by_relid(chunk_relid, true);
+
+	/* Throw error if chunk has invalid status for operation */
+	ts_chunk_validate_chunk_status_for_operation(chunk_state_after_lock->table_id,
+												 chunk_state_after_lock->fd.status,
+												 CHUNK_COMPRESS,
+												 true);
```

Comment:
It might be a good idea to change this to `NOTICE` with the next major release. We currently use `ERROR` in many places. In my opinion we would need to change this everywhere to be consistent.

Reply:
SGTM.